modules/social_features/social_group/social_group.install

```diff
@@ -91,10 +91,15 @@ function _social_group_get_permissions($role) {
   $permissions['authenticated'] = array_merge($permissions['anonymous'], [
     'create open_group group',
     'create closed_group group',
-    'create public_group group',
     'view group stream page',
   ]);
 
+  $config = \Drupal::config('entity_access_by_field.settings');
+  $disable_public_visibility = $config->get('disable_public_visibility');
+  if ($disable_public_visibility === 0) {
+    $permissions['authenticated'][] = 'create public_group group';
+  }
+
```

Comment:
Good point, but I don't think we should be doing that in distro. Because we usually don't change permissions for existing installations; also we can mention this in the release notes.